### PR TITLE
Qt 6 Survey 20210113

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

Introduce Qt version 6.

Package(s) Affected
-------------------
```
qt-6
libb2
```

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-libs/libb2/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=libb2
+PKGSEC=libs
+PKGDES="A C library providing BLAKE2b, BLAKE2s, BLAKE2bp, BLAKE2sp hash functions"
```
Comment (MingcongBai) [inline extra-libs/libb2/autobuild/defines]: ..., and BLAKE2sp *

File: extra-libs/qt-6/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+BUILDDEP="gtk-3 mariadb openssl postgresql unixodbc gperf ninja cmake"
+
+ABTYPE=cmakeninja
+# To query possible build options, you may want to check following files in
+# Qt 6 source tree: qtbase/configure, qtbase/config_help.txt,
+# qtbase/cmake/configure-cmake-mapping.md and
+# qtbase/cmake/QtProcessConfigureArgs.cmake.
+# It's also helpful to refer to build options used by other distributions.
+CMAKE_AFTER="-DINSTALL_BINDIR=bin \
+             -DINSTALL_DOCDIR=share/doc/qt6 \
```
Comment (MingcongBai) [inline extra-libs/qt-6/autobuild/defines]: Should be `share/doc/qt-6`. This was a mistake we made in Qt 5, and wouldn't be easily fixed unless we run a survey for all the packages containing `/usr/share/doc/qt5`.

File: extra-libs/qt-6/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+             -DINSTALL_HEADERDIR=include/qt6 \
+             -DINSTALL_ARCHDATADIR=lib/qt6 \
+             -DINSTALL_MKSPECSDIR=lib/qt6/mkspecs \
+             -DINSTALL_DATADIR=share/qt6 \
+             -DINSTALL_SYSCONFDIR=/etc/xdg \
+             -DINSTALL_EXAMPLESDIR=share/doc/qt6/examples \
```
Comment (MingcongBai) [inline extra-libs/qt-6/autobuild/defines]: Ditto above.

File: extra-libs/qt-6/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+             -DQT_FEATURE_icu=ON \
+             -DQT_FEATURE_journald=ON \
+             -DQT_FEATURE_libinput=ON \
+             -DQT_FEATURE_libproxy=ON \
+             -DQT_FEATURE_vulkan=ON"
+
+CMAKE_AFTER__AMD64=" \
+             ${CMAKE_AFTER} \
+             -DQT_FEATURE_sse2=ON \
+             -DQT_FEATURE_sse3=ON"
```
Comment (MingcongBai) [inline extra-libs/qt-6/autobuild/defines]: Our AMD64 port uses the SSE2 baseline.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by WhiredPlanck** [comment]
There is a fatal typo that wrong date is used in the branch name, so I closed this and will open another PR.

**Pull request closed without merging**